packages/aws-cdk-lib/aws-logs/lib/data-protection-policy.ts

```diff
@@ -0,0 +1,375 @@
+import { Stack } from '../../core';
+import { Construct } from 'constructs';
+/**
+ * Represents a data protection policy in a log group.
+ */
+export class DataProtectionPolicy {
```

Comment:
Can you `extends Construct`?

Reply:
This is a property on `LogGroup`, this shouldn't extend Construct/Resource.

Reply:
Why not? Constructs are not limited to resources.

Reply:
Yeah on second look I was thinking about extending construct because the constructor took a `scope` argument. I think we should modify this API a bit to be something like.

```ts
class DataProtectionPolicy {
  constructor(props: DataProtectionPolicyProps) {
    ...
  }
  public bind(scope: Construct): DataProtectionPolicyConfig {
    ...
  }
}

```

and then
```ts
new CfnLogGroup(this, 'Resource', {
  ...,
  dataProtectionPolicy: props.dataProtectionPolicy.bind(this),
```

Reply:
Updated to match.